numpy/_core/src/multiarray/_multiarray_tests.c.src

```diff
@@ -496,7 +496,7 @@ static PyObject *
 create_custom_field_dtype(PyObject *NPY_UNUSED(mod), PyObject *args)
 {
     PyArray_DescrProto proto;
-    PyArray_Descr *dtype;
+    _PyArray_LegacyDescr *dtype;  /* Is checked for void, so legacy is OK */
```

Comment:
I'd tend to go the other way around, and just recast or define a new variable when you need the legacy one. But this is fine as is too.